src/BoundaryConditions/fill_halo_regions_open.jl

```diff
@@ -1,39 +1,109 @@
+@kernel function _fill_south_and_north_open_halo!(c, south_bc, north_bc, loc, grid, args)
     i, k = @index(Global, NTuple)
-    @inbounds v[i, j_boundary, k] = getbc(bc, i, k, grid, args...)
+    _fill_south_open_halo!(i, k, grid, c, south_bc, loc, args...)
+    _fill_north_open_halo!(i, k, grid, c, north_bc, loc, args...)
 end
 
-@kernel function set_bottom_or_top_w!(w, k_boundary, bc, grid, args) 
+@kernel function _fill_bottom_and_top_open_halo!(c, bottom_bc, top_bc, loc, grid, args)
     i, j = @index(Global, NTuple)
-    @inbounds w[i, j, k_boundary] = getbc(bc, i, j, grid, args...)
+    _fill_bottom_open_halo!(i, j, grid, c, bottom_bc, loc, args...)
+       _fill_top_open_halo!(i, j, grid, c, top_bc,    loc, args...)
 end
 
-@inline   fill_west_halo!(u, bc::OBC, kernel_size, offset, loc, arch, grid, args...; kwargs...) = launch!(arch, grid, KernelParameters(kernel_size, offset), set_west_or_east_u!,   u,           1, bc, grid, Tuple(args); kwargs...)
-@inline   fill_east_halo!(u, bc::OBC, kernel_size, offset, loc, arch, grid, args...; kwargs...) = launch!(arch, grid, KernelParameters(kernel_size, offset), set_west_or_east_u!,   u, grid.Nx + 1, bc, grid, Tuple(args); kwargs...)
-@inline  fill_south_halo!(v, bc::OBC, kernel_size, offset, loc, arch, grid, args...; kwargs...) = launch!(arch, grid, KernelParameters(kernel_size, offset), set_south_or_north_v!, v,           1, bc, grid, Tuple(args); kwargs...)
-@inline  fill_north_halo!(v, bc::OBC, kernel_size, offset, loc, arch, grid, args...; kwargs...) = launch!(arch, grid, KernelParameters(kernel_size, offset), set_south_or_north_v!, v, grid.Ny + 1, bc, grid, Tuple(args); kwargs...)
-@inline fill_bottom_halo!(w, bc::OBC, kernel_size, offset, loc, arch, grid, args...; kwargs...) = launch!(arch, grid, KernelParameters(kernel_size, offset), set_bottom_or_top_w!,  w,           1, bc, grid, Tuple(args); kwargs...)
-@inline    fill_top_halo!(w, bc::OBC, kernel_size, offset, loc, arch, grid, args...; kwargs...) = launch!(arch, grid, KernelParameters(kernel_size, offset), set_bottom_or_top_w!,  w, grid.Nz + 1, bc, grid, Tuple(args); kwargs...)
-
-@inline   _fill_west_halo!(j, k, grid, c, bc::OBC, loc, args...) = @inbounds c[1, j, k]           = getbc(bc, j, k, grid, args...)
-@inline   _fill_east_halo!(j, k, grid, c, bc::OBC, loc, args...) = @inbounds c[grid.Nx + 1, j, k] = getbc(bc, j, k, grid, args...)
-@inline  _fill_south_halo!(i, k, grid, c, bc::OBC, loc, args...) = @inbounds c[i, 1, k]           = getbc(bc, i, k, grid, args...)
-@inline  _fill_north_halo!(i, k, grid, c, bc::OBC, loc, args...) = @inbounds c[i, grid.Ny + 1, k] = getbc(bc, i, k, grid, args...)
-@inline _fill_bottom_halo!(i, j, grid, c, bc::OBC, loc, args...) = @inbounds c[i, j, 1]           = getbc(bc, i, j, grid, args...)
-@inline    _fill_top_halo!(i, j, grid, c, bc::OBC, loc, args...) = @inbounds c[i, j, grid.Nz + 1] = getbc(bc, i, j, grid, args...)
+# fallback for non-open boundary conditions
```

Comment:
```suggestion
# Fallback for open boundary conditions:
```

I think this comment is wrong, it's reversed with the one below. Also it's better to use full sentences / capitalization for any English

Reply:
This is the generic fallback for `_fill_X_open_halo!` and below is the fallback for open boundaries on `_fill_X_halo!`, although I'll delete the next set because that's just the default fallback.